airflow/models.py

```diff
@@ -3608,12 +3608,31 @@ def set(
             execution_date,
             task_id,
             dag_id,
-            session=None):
+            session=None,
+            enable_pickling=None):
         """
         Store an XCom value.
+        TODO: "pickling" has been deprecated and JSON is preferred. "pickling" will be removed in Airflow 2.0.
+        :param enable_pickling: If pickling is not enabled, the XCOM value will be parsed as JSON instead.
+        :return: None
         """
         session.expunge_all()
 
+        if enable_pickling is None:
+            enable_pickling = configuration.getboolean('core', 'enable_pickling')
+
+        if enable_pickling:
+            value = pickle.dumps(value)
```

Comment:
This would be a good place for something like:

```python

            warnings.warn(
                "XComs are no longer automatically pickled... etc. etc. ",
                DeprecationWarning)
```

Which is how deprecated features are marked elsewhere in Airflow